base/error.jl

```diff
@@ -18,24 +18,83 @@
 
 ## native julia error handling ##
 
+"""
+    error(message::AbstractString)
+
+Raise an `ErrorException` with the given message.
+"""
 error(s::AbstractString) = throw(ErrorException(s))
 error(s...) = throw(ErrorException(Main.Base.string(s...)))
 
+"""
+    rethrow([e])
+
+Throw an object without changing the current exception backtrace. The default argument is
+the current exception (if called within a `catch` block).
+"""
 rethrow() = ccall(:jl_rethrow, Bottom, ())
 rethrow(e) = ccall(:jl_rethrow_other, Bottom, (Any,), e)
+
+"""
+    backtrace()
+
+Get a backtrace object for the current program point.
+"""
 backtrace() = ccall(:jl_backtrace_from_here, Array{Ptr{Void},1}, (Int32,), false)
+
+"""
+    catch_backtrace()
+
+Get the backtrace of the current exception, for use within `catch` blocks.
+The backtrace is represented as an `Array{Ptr{Void}}` of addresses, so [`showerror`](:func:`showerror`)
+may be helpful for displaying the backtrace in a human-readable format.
+
+# Example
+
+```jldoctest
+julia> f(x) = try
+           sqrt(x)
+       catch y
+           bt = catch_backtrace()
+           showerror(STDOUT, y, bt)
+       end;
+
+julia> f(-1)
+DomainError:
+ in f(::Int64) at ./REPL[24]:2
```

Comment:
>That should be included since it at least means what tony said won't work.

This doctest fails even when I include the full output listed for "got".

After formatting this doctest several different ways, the only solutions I found involved adding calls to `println()`. I suspect there may be a bug in the way the doctest system is parsing this code block, but I did not find anything suspicious after going through what I could find of the documentation code. If there aren't other suggestions for things I might try, then I think I will move on to other fronts. Please let me know if you'd like me to convert this doctest to a regular code block or make some other modification. Thanks! :)